templates/build.and.release.scripted.yml

```diff
@@ -0,0 +1,213 @@
+# This template is designed for builds using the Endjin.RecommendedPractices.Build module:
+#   https://www.powershellgallery.com/packages/Endjin.RecommendedPractices.Build
+#   https://github.com/endjin/Endjin.RecommendedPractices.Build
+# The majority of the build process is executed by a `build.ps1` script in the root of the
+# project.
+#
+# With scripted builds, many of the extensibility points exposed by `build.and.release.yml`
+# (e.g. `preBuild`, `postBuild`, `preRunExecutableSpecs`, etc.) are now supplied by the
+# build script, and not this Azure DevOps template.
+# However, there are still some extensibility points specific to the Azure DevOps pipeline
+# execution.
+parameters:
+  preCustomEnvironmentVariables: []
+  postCustomEnvironmentVariables: []
+  prePublishCodeCoverageReport: []
+  postPublishCodeCoverageReport: []
+  preCopyNugetPackages: []
+  postCopyNugetPackages: []
+  prePublishReleaseArtifacts: []
+  postPublishReleaseArtifacts: []
+  preCreateGitHubRelease: []
+  postCreateGitHubRelease: []
+  prePublishNugetPackages: []
+  postPublishNugetPackages: []
+  vmImage: ''
+  service_connection_nuget_org: ''
+  service_connection_github: ''
+  netSdkVersion: '6.x'
+
+jobs:
+- job: Build
+  pool:
+    vmImage: ${{ parameters.vmImage }}
+
+  variables:
+    BuildConfiguration: 'Release'
+    DOTNET_CLI_TELEMETRY_OPTOUT: 1
+    DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+
+    # We have dependencies on the following Environment Variables:
+    # BUILD_REPOSITORY_NAME
+
+    # We have dependencies on the following Build Variables:
+    # Endjin_Service_Connection_NuGet_Org
+    # Endjin_Service_Connection_GitHub
+    # Endjin_Slack_ReleasesWebhookUri
+    # Endjin.ForcePublish
+  steps:
+  - task: UseDotNet@2
+    displayName: 'Install .NET SDK ${{ parameters.netSdkVersion }}'
+    inputs:
+      packageType: sdk
+      version: ${{ parameters.netSdkVersion }}
+      installationPath: $(Agent.ToolsDirectory)/dotnet
+
+  - ${{ parameters.preCustomEnvironmentVariables }}
+
+  - powershell: |
+      Write-Host "##vso[task.setvariable variable=Endjin_Repository_Name]$Env:BUILD_REPOSITORY_NAME"
+    displayName: 'Set Build Variables from Environment'
+
+  - ${{ parameters.postCustomEnvironmentVariables }}
+
+  - powershell: |
+      Write-Host "Initializing $(Build.ArtifactStagingDirectory)/Release"
+      New-Item -Path $(Build.ArtifactStagingDirectory) -Name "Release" -ItemType "directory"
+    displayName: 'Initialize Artifact Staging Release Directory'
+
+  # Useful for debugging purposes
+  - powershell: 'gci Env:'
+    condition: or(variables['Endjin.BuildDiagnostics'], variables['Endjin.ShowEnvironment'])
+    displayName: 'Print Environment Variables'
+
+  - task: PowerShell@2
+    inputs:
+      targetType: 'filePath'
+      filePath: "$(Build.SourcesDirectory)/build.ps1"
```

Comment:
I think we'll need to pass at least the `Configuration` and `BuildRepositoryUri` parameters to the build script.

Perhaps also `CoverageDir` to align with the `PublishCodeCoverageResults` task later on?

`LogLevel` might be occasionally useful when troubleshooting?

Reply:
Thanks! I've now added these changes. Build coverage appears to be working on a test build.